flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -2729,11 +2727,11 @@ object JobManager {
   def getRemoteJobManagerAkkaURL(config: Configuration) : String = {
     val (protocol, hostname, port) = TaskManager.getAndCheckJobManagerAddress(config)
 
-    var hostPort: InetSocketAddress = null
+    var hostPort: InetSocketAddress = new InetSocketAddress(hostname, port)
```

Comment:
Does this change alter anything compared to before? At a first glance, both construct the same `InetSocketAddress` and both needs to hostname to be resolvable.
Is it important that the `InetSocketAddress` contains the hostname, rather than the IP address?